waiter/src/waiter/process_request.clj

```diff
@@ -858,6 +858,36 @@
             (utils/data->error-response request)))
       (handler request))))
 
+(defn wrap-maintenance-mode
+  "Check if a service's token is in maintenance mode and immediately return a 503 response
+  with a custom message if specified. Check if x-waiter-maintenance header is set and return
+  400 response because the header is not supported."
+  [handler]
+  (fn [{{:keys [token-metadata token waiter-headers service-description-template]} :waiter-discovery
+        {:keys [service-id]} :descriptor
+        :as request}]
+    (let [maintenance (get token-metadata "maintenance")
+          response-map {:maintenance maintenance
+                        :token token
+                        :service-id service-id}]
+      (cond (contains? waiter-headers "x-waiter-maintenance")
+            (do
+              (log/info "x-waiter-maintenance is not supported as an on-the-fly header"
+                        {:service-description service-description-template :token token})
+              (-> {:message "Maintenance parameter is not supported for on-the-fly headers"
+                   :status http-400-bad-request
+                   :waiter-headers waiter-headers}
+                  (utils/data->error-response request)))
+            (some? maintenance)
+            (do
+              (log/info "token is in maintenance mode" response-map)
+              (meters/mark! (metrics/service-meter service-id "response-rate" "error" "maintenance"))
+              (-> {:details response-map,
```

Comment:
Please drop the commas. As a stylistic guideline, we try and avoid use of commas where possible.